File: python/purehtml/PureHTML.py
from typing import Any

from purehtml.backend.backend import PureHTMLInitializer
from purehtml.configs.ExtractParams import ExtractParams


def extract(config, html, url) -> Any:
    # Initialize the backend and load the HTML
    doc = PureHTMLInitializer().load(html)

    extract_params = ExtractParams(
        document=doc, nodes=[], url=url, element_already_matched=False
    )

    return config.extract(extract_params)

File: java/src/main/java/org/purescraps/purehtml/PureHTML.java
package org.purescraps.purehtml;

import org.purescraps.purehtml.backend.JsoupBackend;
import org.purescraps.purehtml.backend.PureHTMLDocument;
import org.purescraps.purehtml.configs.Config;
import org.purescraps.purehtml.interfaces.ExtractParams;

import java.util.ArrayList;

public class PureHTML {

    public static Object extract(Config config, String html, String url) {

        PureHTMLDocument doc = JsoupBackend.load(html);
        ExtractParams extractParams = new ExtractParamsBuilder()
                .setDocument(doc)
                .setNode(new ArrayList<>())
                .setUrl(url)
                .setElementAlreadyMatched(false)
                .build();

        return config.extract(extractParams);

    }

}
